docs/source/deprecation.md

```diff
@@ -1,14 +1,164 @@
+in the report, ... But the rest of the functionality is the same as before,
+until the entity is removed completely.
+
+## What is the deprecation process for leapp
+
+In case some public public functions or models are to be removed for any
+reason, these need to be marked as deprecated before the removal (if possible).
+In our projects, the deprecation will be applied only for functions and models
+that have been introduced in an official release (IOW, a functionality that
+has been merged into the upstream, but has been removed before the release
+or was marked as experimental all the time, is going to be removed without
+the deprecation state). The time period during which the deprecated entity
+won't be removed is at least 6 months. That doesn't mean we will remove
+everything deprecated immadiately after the 6 months, but it's to be expected
+that it will be dropped anytime between 6 and 12 months since the deprecation.
+
+In case of issues, deprecated functionality is not going to be fixed since
+it is deprecated (unless it is fixed e.g. as a side-effect of another
+problem fix).
+
+## How does leapp informs about deprecated functionality?
+
+Mainly via release notes and changelogs. In the official leapp related projects
+(especially leapp and leapp-repository) the OAMG team take care about release
+notes to ensure they contain all data about dropped and deprecated
+functionality.
+
+Additionally, when using leapp or snactor, user is notified via messages about
+deprecated entities in **limited cases** (see below). In case of the leapp
+utility, such messages are presented inside the generated reports. In case
+of the snactor utility, the information message is printed in the console
+output in the end of the snactor execution. See examples in this page for
+detail.
+
+But be aware of limitations of our detection in leapp and snactor utilities.
+Our detection mechanisms are not bulletproof. Here is list of some examples,
+when our detection does not work well:
+
+- when an actor a message of a deprecated model, the usage of the deprecated
+  model is detected only if the message has been produced before; IOW, in
+  case the message X is produced only on machines with a specific package,
+  the deprecated usage is not detected on machines without the packagw
+- when a message of a deprecated model is consumed, the message points
+  internals of the leapp framework, instead of place where the message is
+  consumed, e.g.
+  ```
+  Usage of deprecated Model "Foo" @ /usr/lib/python2.7/site-packages/leapp/models/__init__.py:113
+  Near:         return cls(init_method='to_model', **data)
+  ```
+- a deprecated class (including model) is reported only when an instance of the
+  class (object) is created during the runtime
+- use of a deprecated function/method is reported only when the function is
+  called
+
+There are probably other cases, but these are the most common, that could
+confuse people.
```

Comment:
let's remove this as it sounds, basically, not reliable. If we face other problems, we have to update the documentation

Reply:
For almost sure there could be others. But do we want to write all cases? The purpose of the message above is to show, that it's just additional level to inform people about deprecated stuff, that it will not inform you in 100% cases about all deprecated stuff and still, there is the only source of truth (changelogs, release notes).